addons/mail/models/discuss/discuss_channel.py

```diff
@@ -739,10 +752,12 @@ def _find_or_create_member_for_self(self):
         if member:
             return member
         if not self.env.user._is_public():
-            return self.add_members(partner_ids=self.env.user.partner_id.ids)
+            return self.add_members(
+                partner_ids=self.env.user.partner_id.ids, post_joined_message=not self.parent_channel_id
```

Comment:
Do we ever want to post join message if parent_channel_id is set?
Otherwise this condition could directly be handled in add_member.

Reply:
Removed the custom logic to get closed from other channels (leave/join messages are posted like for the other channels)